scripts/nn/test/grad_check.dml

```diff
@@ -951,6 +952,45 @@ cross_entropy_loss = function() {
   }
 }
 
+cross_entropy_loss2d = function() {
+  /*
+   * Gradient check for the 2D cross-entropy loss function.
+   */
+  print("Grad checking the 2D cross-entropy loss function.")
+
+  # Generate data
+  N = 3  # num examples
+  C = 10  # num targets
+  Hin = 28  # example height
+  Win = 28  # example width
```

Comment:
Using 28 was great while creating this PR to ensure that there was no stability issues, but I will reduce it to 5 when I merge just to reduce the amount of time it takes to run the test.